### ISSUE TITLE
Surface Book 2: Lid closing/opening event not detected

### ISSUE THREAD

**Issue by deb75** [jakeday/linux-surface#146]
Hello,

I noticed that closing/opening the lid did not trigger the expected action.

Closing and opening the lid with `acpi_listen` did not show anything.

But in logs, this shows up : 
```[  128.741415] ACPI Error: No handler for Region [OR01] (00000000a595f871) [GenericSerialBus] (20170831/evregion-166)
[  128.741420] ACPI Error: Region GenericSerialBus (ID=9) has no handler (20170831/exfldio-299)
[  128.741422] ACPI Error: Method parse/execution failed \_SB._SAN.RQSX, AE_NOT_EXIST (20170831/psparse-550)
[  128.741426] ACPI Error: Method parse/execution failed \_SB._SAN.RQST, AE_NOT_EXIST (20170831/psparse-550)
[  128.741429] ACPI Error: Method parse/execution failed \_SB.LID0.GBOS, AE_NOT_EXIST (20170831/psparse-550)
[  128.741431] ACPI Error: Method parse/execution failed \_SB.LID0.ULID, AE_NOT_EXIST (20170831/psparse-550)
[  128.741433] ACPI Error: Method parse/execution failed \_GPE._L17, AE_NOT_EXIST (20170831/psparse-550)
[  128.741436] ACPI Exception: AE_NOT_EXIST, while evaluating GPE method [_L17] (20170831/evgpe-646)
[  131.121198] ACPI Error: No handler for Region [OR01] (00000000a595f871) [GenericSerialBus] (20170831/evregion-166)`
[  131.121203] ACPI Error: Region GenericSerialBus (ID=9) has no handler (20170831/exfldio-299)
[  131.121205] ACPI Error: Method parse/execution failed \_SB._SAN.RQSX, AE_NOT_EXIST (20170831/psparse-550)
[  131.121209] ACPI Error: Method parse/execution failed \_SB._SAN.RQST, AE_NOT_EXIST (20170831/psparse-550)
[  131.121211] ACPI Error: Method parse/execution failed \_SB.LID0.GBOS, AE_NOT_EXIST (20170831/psparse-550)
[  131.121213] ACPI Error: Method parse/execution failed \_SB.LID0.ULID, AE_NOT_EXIST (20170831/psparse-550)
[  131.121216] ACPI Error: Method parse/execution failed \_GPE._L17, AE_NOT_EXIST (20170831/psparse-550)
[  131.121218] ACPI Exception: AE_NOT_EXIST, while evaluating GPE method [_L17] (20170831/evgpe-646)
```

So the event is detected but not properly handled by the acpi system. Is it part of the overall acpi bug (power buttons, battery stats) ?

Tested on SB2 + dGPU + recompiled surface kernel 4.15.16

**Comment by deb75** [comment]
@jakeday Hello, could you progress with the acpi notify module ? If you need it to be tested on a SB2 13", I can use mine, feel free to ask for any help, regards

**Comment by calvous** [comment]
Just some input on that Lid problem. I have a Surface Pro 1796 (sometimes identified as V). While the battery is not detected and the AC (ADP1) is always in state off-line, the Lid seems to work.
``
[suchus@PSYCHO ~]$ uname -r
4.16.14-2-MANJARO
[suchus@PSYCHO ~]$ dmesg|grep Lid
[    0.456820] input: Lid Switch as /devices/LNXSYSTM:00/LNXSYBUS:00/PNP0C0D:00/input/input0
[    0.456827] ACPI: Lid Switch [LID0]
[suchus@PSYCHO ~]$ journalctl | grep LID | tail -4
Jun 11 07:27:27 PSYCHO root[1063]: LID closed
Jun 11 07:27:32 PSYCHO root[1304]: LID opened
Jun 11 07:32:30 PSYCHO root[1541]: LID closed
Jun 11 07:32:36 PSYCHO root[1726]: LID opened
``

**Comment by qzed** [comment]
I'm currently trying to implement support for the lid state. If you want to help me, I need to collect a bit of information regarding some signals. Have a look [here](https://github.com/qzed/surfacebook2-linux-mshw0153/issues/1) for the details.

**Comment by limyongsheng97** [comment]
Hello just curious if this issue had been resolved ? Seems like closing the lid is still non-responsive even after installing the latest version of @jakeday 's kernel (thank you for your efforts by the way, much appreciated)

I am running GNOME tweaks on Ubuntu 18.04 LTS on a SB2 13.5 and the laptop does not go to sleep when the option "Suspend when laptop lid is closed" is enabled and the lid is closed.

This lid issue is related to :
What's NOT Working
Suspend (uses Connected Standby which is not supported yet) ?

**Comment by qzed** [comment]
@limyongsheng97 The lid event stuff will be implemented in combination with #28 (the embedded controller required for battery information also handles some other things), on which I'm currently working on. Connected Standby still won't work after this, however.

**Comment by jakeday** [comment]
This has been merged in and is now working.
